repocreation/createRepo.go

```diff
@@ -0,0 +1,212 @@
+	if err != nil {
+		log.ErrorCtx(ctx, errors.Wrap(err, "Adding protection, require signatures failed on branch master"), log.Data{"resp": resp})
+	}
+	_, resp, err = client.Repositories.RequireSignaturesOnProtectedBranch(ctx, org, repoName, "develop")
+	if err != nil {
+		log.ErrorCtx(ctx, errors.Wrap(err, "Adding protection, require signatures failed on branch develop"), log.Data{"resp": resp})
+	}
+	return err
+}
+
+// setDevelopAsDefaultBranch will set the develop branch as the repositories default branch
+func setDevelopAsDefaultBranch(client *github.Client, ctx context.Context, repoName string, repo *github.Repository) error {
+	repo.AutoInit = nil
+	_, resp, err := client.Repositories.Edit(ctx, org, repoName, repo)
+	if err != nil {
+		log.ErrorCtx(ctx, errors.Wrap(err, "Failed to set develop as the default branch"), log.Data{"resp": resp})
+	}
+	return err
+}
+
+// createDevelopBranch will create a new branch named /develop, based on the last ref on master branch
+func createDevelopBranch(client *github.Client, ctx context.Context, repoName string) error {
+	ref, resp, err := client.Git.GetRef(ctx, org, repoName, "heads/master")
+	if err != nil {
+		log.ErrorCtx(ctx, errors.Wrap(err, "Get Reference to master commit failed"), log.Data{"resp": resp})
+		return err
+	}
+	developBranch := "heads/develop"
+	ref.Ref = &developBranch
+
+	_, resp, err = client.Git.CreateRef(ctx, org, repoName, ref)
+	if err != nil {
+		log.ErrorCtx(ctx, errors.Wrap(err, "Create Reference to new develop branch failed"), log.Data{"resp": resp})
+		return err
+	}
+	return nil
+}
+
+// createRepo makes a call via the gitHubAPI to generate a new repository
+func createRepo(client *github.Client, ctx context.Context, repo *github.Repository) error {
+	newRepo, _, err := client.Repositories.Create(ctx, org, repo)
+	fmt.Println(newRepo)
+	if err != nil {
+		log.ErrorCtx(ctx, errors.Wrap(err, "repo creation failed"), nil)
+		return err
+	}
+	return nil
+}
+
+// getConfigurationsForNewRepo gets required configuration information from the end user
+func getConfigurationsForNewRepo() (accessToken, userHandle, repoName, repoDescription string) {
+	accessToken = promptForInput("Please provide your user access token, to create one follow this guide https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line")
+	userHandle = promptForInput("Please provide your github handle/username")
+	repoName = promptForInput("Please provide the full name for the new repository")
+	repoDescription = promptForInput("Please provide a description for the repository")
+	return accessToken, userHandle, repoName, repoDescription
+}
+
+// promptForInput gives a user a message and expect input to be provided
+func promptForInput(prompt string) string {
+	fmt.Println(prompt)
+	reader := bufio.NewReader(os.Stdin)
+	input, _ := reader.ReadString('\n')
+	return strings.TrimSuffix(input, "\n")
```

Comment:
You can use a Scanner for this: https://golang.org/pkg/bufio/#example_Scanner_lines